fees/x402/index.ts

```diff
@@ -0,0 +1,210 @@
+];
+
+const SENDER_SET = new Set(FACILITATOR_SENDERS_BASE.map((s) => s.address.toLowerCase()));
+const SINCE_BY_SENDER = new Map(
+  FACILITATOR_SENDERS_BASE.map((s) => [s.address.toLowerCase(), s.since || 0]),
+);
+
+const START = "2024-12-05"; // First x402rs facilitator
+
+const fetch = async (options: FetchOptions) => {
+  const dailyVolume = options.createBalances();
+
+  // 1) Get AuthorizationUsed events from USDC(Base)
+  const authLogs: any[] = await options.getLogs({
+    target: USDC_BASE,
+    eventAbi: AUTH_USED_EVENT
+  });
+
+  if (!authLogs?.length) {
+    return {
+      dailyVolume,
+      dailyFees: 0,
+      dailyRevenue: 0,
+    };
+  }
+
+  const txHashes = [...new Set(authLogs.map((l) => l.transactionHash))];
+
+  // 2) Preferred path: receipts (filter sender+since; sum USDC Transfers)
+  const receipts = (options as any).getTxReceipts
+    ? await (options as any).getTxReceipts(txHashes)
+    : [];
+
+  if (receipts.length) {
+    for (const r of receipts) {
+      if (!r) continue;
+      const sender = (r.from || "").toLowerCase();
+      if (!SENDER_SET.has(sender)) continue;
+
+      const blockTimeMs = (r.blockTimestamp ?? 0) * 1000;
+      const sinceMs = SINCE_BY_SENDER.get(sender) ?? 0;
+      if (sinceMs && blockTimeMs && blockTimeMs < sinceMs) continue;
+
+      // Note: We don't filter by method selector - AuthorizationUsed + sender is sufficient.
+      // This avoids missing x402 transactions that go through wrappers/variants.
+
+      for (const log of r.logs || []) {
+        const addr = (log.address || "").toLowerCase();
+        if (addr !== USDC_BASE.toLowerCase()) continue;
+        if (!log.topics || log.topics[0]?.toLowerCase() !== TRANSFER_TOPIC) continue;
+        dailyVolume.add(`base:${USDC_BASE}`, log.data);
+      }
+    }
+    return {
+      dailyVolume,
+      dailyFees: 0,
+      dailyRevenue: 0,
+    };
+  }
+
+  // 3) Fallback: if no receipts, use getTransactions to filter sender
+  if ((options as any).getTransactions) {
+    const txs: any[] = await (options as any).getTransactions(txHashes);
+    const allowedHashes = new Set<string>();
+    for (const tx of txs) {
+      const sender = (tx.from || "").toLowerCase();
+      if (!SENDER_SET.has(sender)) continue;
+      // Can't check since without timestamps; if runner exposes receipts "per hash", try 1 call
+      if ((options as any).getTxReceipts) {
+        const [r] = await (options as any).getTxReceipts([tx.hash]);
+        const blockTimeMs = (r?.blockTimestamp ?? 0) * 1000;
+        const sinceMs = SINCE_BY_SENDER.get(sender) ?? 0;
+        if (sinceMs && blockTimeMs && blockTimeMs < sinceMs) continue;
+      }
+      allowedHashes.add(tx.hash);
+    }
+
+    // Intersect with USDC Transfers to get amounts
+    if (allowedHashes.size) {
+      const TRANSFER_EVENT = "event Transfer(address indexed from, address indexed to, uint256 value)";
+      const transferLogs: any[] = await options.getLogs({
```

Comment:
pulling all transfers of usdc on base is gonna be extremely slow and resource intensive